spec/crypto/flow-encryption/dkg.md

```diff
@@ -0,0 +1,136 @@
+# Distributed Key Generation
+
+A prerequisite for flow encryption in Penumbra is some distributed key
+generation algorithm. Our [threshold encryption](./threshold-encryption.md)
+scheme uses ElGamal and operates over the `decaf377` group, thus we must choose
+a DKG which operates over `decaf377` outputs a `decaf377` public key. 
+
+## Desired Properties
+
+### Minimal Round Complexity
+
+The DKG must be run by all validators at the start of every epoch. Ideally, the
+DKG should be able to run in over a single block, such that there is no delay
+between the start of the epoch and when users can delegate in the staking
+system or trade on ZSwap. As such, for optimal UX and simplicity of
+implementation, the optimal DKG for our scheme should have minimal round
+complexity. Each round of communication can occur in the
+[ABCI++](https://github.com/tendermint/spec/blob/master/rfc/004-abci%2B%2B.md)
+vote extension phase. Since Tendermint votes already impose an `O(n^2)`
+communication complexity for each block, minimal communication complexity is
+not important for our DKG.
+
+### Verifiability and Security Under Rogue-Key Attacks
+
+In addition to minimal round complexity, our DKG must be *verifiable*:
+participants (validators) must be able to verify that counterparty participants
+(other validators) are contributing to the DKG honestly, without the use of a
+trusted dealer. This can be achieved using something similar to [Feldman's
+Verifiable Secret Sharing][feldman] protocol, where each participant shares a
+commitment to their share which is visable to all other participants. In
+addition, our DKG must be able to tolerate *rogue-key attacks*: that is, it
+must tolerate the instance where a validator maliciously chooses their share
+based on the value of the other validator's shares in order to cancel out other
+validator's keyshares and gain unilateral control over the resulting DKG key.
+One way this can be prevented is by each validator producing a proof of
+knowledge of their secret share.
+
+### Robustness
+
+An additional property that our DKG should have is that of *robustness*. The
+DKG should be able to tolerate a byzantine threshold of validators
+intentionally refusing to participate in the DKG round, or intentionally
+contributing malformed shares during DKG execution, without requiring a full
+restart of the DKG protocol. This is due to DoS concerns: with a naive,
+non-robust implementation, a single malicious validator could potentially
+indefinitely delay the beginning of an epoch by refusing to participate in DKG
+or by contributing invalid shares.
+
+
+## A Survey of Existing DKGs
+
+[Gurkan, Jovanovic, Maller, Meiklejohn, Stern, and Tomescu][GJMMST] present a survey of DKG instantiations with their communication complexity, public verifiablity, round complexity, prover, and verifier costs:
+
+![](./dkg-table.png)
```

Comment:
I think it might make sense to re-code the table as a Markdown table at some point